processor/src/operations/crypto_ops.rs

```diff
@@ -148,7 +152,7 @@ impl Process {
         // specified depth. if the new node is the root of a tree, this instruction will append the
         // whole sub-tree to this node.
         let path: MerklePath =
-            host.advice_provider_mut().update_operand_stack_merkle_node(self.into())?;
+            update_operand_stack_merkle_node(host.advice_provider_mut(), self.into())?;
```

Comment:
Same comment as the one above: I'd move the code from `update_operand_stack_merkle_node()` here and get rid of that function.